basis/http/client/client.factor

```diff
@@ -6,17 +6,40 @@ http.parsers io io.crlf io.encodings io.encodings.ascii
 io.encodings.binary io.encodings.iana io.encodings.string
 io.files io.pathnames io.sockets io.timeouts kernel locals math
 math.order math.parser mime.types namespaces present sequences
-splitting urls vocabs.loader ;
+splitting urls vocabs.loader combinators environment ;
 IN: http.client
 
 ERROR: too-many-redirects ;
 
+: success? ( code -- ? ) 200 299 between? ;
+
+ERROR: download-failed response ;
+
+: check-response ( response -- response )
+    dup code>> success? [ download-failed ] unless ;
+
 <PRIVATE
 
+: authority-uri ( request -- str )
+    [ host>> ] [ port>> number>string ] bi ":" glue ;
```

Comment:
It will throw an exception on f number>string... That's not supposed to happen because <client-request> calls ensure-port.
Also, I just noticed the stack effect name is wrong, this needs a url..